app/jobs/shipit/update_github_last_deployed_ref_job.rb

```diff
@@ -0,0 +1,37 @@
+module Shipit
+  class UpdateGithubLastDeployedRefJob < BackgroundJob
+    queue_as :default
+
+    DEPLOY_PREFIX = 'shipit-deploy'.freeze
+
+    def perform(stack)
+      stack_sha = stack.deploys.last.until_commit.sha
+      environment = stack.environment
+      stack_ref = create_full_ref(environment)
+      client = Shipit.github.api
+
+      full_repo_name = stack.github_repo_name
+
+      update_or_create_ref(client: client, repo_name: full_repo_name, ref: stack_ref, new_sha: stack_sha)
+    end
+
+    private
+
+    def create_full_ref(stack_environment)
+      [DEPLOY_PREFIX, stack_environment].join("/")
+    end
+
+    def create_ref(client:, repo_name:, ref:, sha:)
+      client.create_ref(repo_name, ref, sha)
+    end
+
+    def update_or_create_ref(client:, repo_name:, ref:, new_sha:)
+      client.update_ref(repo_name, ref, new_sha)
+    rescue Octokit::UnprocessableEntity => e
+      error_msg = e.message
+      if error_msg.include? "Reference does not exist"
+        create_ref(client: client, repo_name: repo_name, ref: ref, sha: new_sha)
+      end
```

Comment:
We should `raise` if we don't recognise the error message.

Reply:
I'm re-raising in the other branch of the conditional now. Let me know if you meant something else.